ERA5/construct_era5_daily_cancities_revised.py

```diff
@@ -0,0 +1,444 @@
+    )
+
+    if "pr" not in dly.data_vars:
+        # Total precip flux in kg m-2 s-1
+        pr = hrly.pr.resample(time="D").sum()
+    else:
+        pr = dly.pr
+
+    pr.attrs.update(
+        standard_name="precipitation_flux",
+        long_name="Mean daily precipitation flux",
+        units="kg m-2 s-1",
+        cell_methods="time: mean within days",
+        description="Total precipitation thickness converted to mass flux using a water density of 1000 kg/m³.",
+    )
+
+    if "evspsblpot" not in dly.data_vars:
+        # Total Potential Evapotranspiration flux in kg m-2 s-1
+        evspsblpot = hrly.evspsblpot.resample(time="D").sum()
+    else:
+        evspsblpot = dly.evspsblpot
+
+    evspsblpot.attrs.update(
+        standard_name="water_potential_evaporation_flux",
+        long_name="Mean daily potential evaporation flux",
+        units="kg m-2 s-1",
+        cell_methods="time: mean within days",
+        description="Total potential evaporation thickness converted to mass flux using a water density of 1000 kg/m³.",
+    )
+
+    if "prsn" not in dly.data_vars:
+        # Total Solid Precipitation flux in kg m-2 s-1
+        prsn = hrly.prsn.resample(time="D").sum()
+    else:
+        prsn = dly.prsn
+
+    snw = hrly.snw.resample(time="D").mean() if "snw" not in dly.data_vars else dly.snw
+    swe = snw / 1000
+    snr = hrly.snr.resample(time="D").mean() if "snr" not in dly.data_vars else dly.snr
+    if "snd" not in dly.data_vars:
+        if "snd" in hrly.data_vars:
+            snd = hrly.snd.resample(time="D").mean()
+        else:
+            snd = snw / snr
+    else:
+        snd = dly.snd
+
+    prsn.attrs.update(
+        standard_name="solid_precipitation_flux",
+        long_name="Mean daily solid precipitation",
+        units="kg m-2 s-1",
+        cell_methods="time: mean within days",
+    )
+    snw.attrs.update(
+        standard_name="surface_snow_amount",
+        long_name="Surface snow amount",
+        units="kg m-2",
+        cell_methods="time: mean within days",
+    )
+    snd.attrs.update(
+        standard_name="surface_snow_thickness",
+        long_name="Snow depth",
+        units="m",
+        cell_methods="time: mean within days",
+    )
+    swe.attrs.update(
+        standard_name='lwe_thickness_of_surface_snow_amount',
+        long_name='Liquid water equivalent of surface snow amount',
+        units='m',
+        cell_methods='time: mean within days'
+    )
+
+    if "uas" not in dly.data_vars:
+        uas = hrly.uas.resample(time="D").mean()
+    else:
+        uas = dly.uas
+
+    if "vas" not in dly.data_vars:
+        vas = hrly.vas.resample(time="D").mean()
+    else:
+        vas = dly.vas
+
+    windmag, _ = xc.atmos.wind_speed_from_vector(uas=hrly.uas, vas=hrly.vas)
+    sfcWind = windmag.resample(time="D").mean()
+    wsgsmax = windmag.resample(time="D").max()
+
+    _, sfcWindfromdir = xc.atmos.wind_speed_from_vector(uas=uas, vas=vas)
```

Comment:
In theory, `uas` and `vas` can't be resampled directly.  See example in xscen : https://github.com/Ouranosinc/xscen/blob/331e2ecc78fdf9cb96ac903319f7371ea7067f47/xscen/extract.py#L470

In summary, the real daily components are reconstructed from the resampled magnitude and the angle of the resampled components.

Reply:
It is not primordial to do it correctly here, but it feels like a low-hanging fruit.